spyrograph/_trochoid.py

```diff
@@ -67,7 +67,12 @@ def __init__(
         self.d = d
         self.thetas = self._validate_theta(thetas, theta_start, theta_stop, theta_step)
         self.origin = origin
-
+   
+        while deci != False:
+            if self.R < 0 or self.r < 0 or self.d < 0:
+                deci = True
+            else:
+                deci = False
```

Comment:
This `while` loop is redundant and could be accomplished with just the `if` statements also we need to `raise` a `ValueError` if the `if` statement is triggered

Something along the lines of:

```python
if self.R < 0 or self.r < 0 or self.d < 0:
    raise ValueError("Negative input parameters were passed. Please only pass positive values")
```

Reply:
In reference to triggering on zero as well like mentioned in #92 this could be 

```python
if self.R <= 0 or self.r <= 0 or self.d <= 0:
    raise ValueError("Negative and/or zero input parameters were passed. Please only pass positive values")
```

This way it will trigger on zero as well

Reply:
Ohk I'll make the changes.